src/host/VtApiRoutines.cpp

```diff
@@ -0,0 +1,885 @@
+        (void)m_pVtEngine->RequestCursor();
+    }
+    return hr;
+}
+
+[[nodiscard]] HRESULT VtApiRoutines::ReadConsoleAImpl(IConsoleInputObject& context,
+                                                      gsl::span<char> buffer,
+                                                      size_t& written,
+                                                      std::unique_ptr<IWaitRoutine>& waiter,
+                                                      const std::string_view initialData,
+                                                      const std::wstring_view exeName,
+                                                      INPUT_READ_HANDLE_DATA& readHandleState,
+                                                      const HANDLE clientHandle,
+                                                      const DWORD controlWakeupMask,
+                                                      DWORD& controlKeyState) noexcept
+{
+    const auto hr = m_pUsualRoutines->ReadConsoleAImpl(context, buffer, written, waiter, initialData, exeName, readHandleState, clientHandle, controlWakeupMask, controlKeyState);
+    // If we're about to tell the caller to wait, let's synchronize the cursor we have with what
+    // the terminal is presenting in case there's a cooked read going on.
+    // TODO: we only need to do this in cooked read mode.
+    if (clientHandle)
+    {
+        m_listeningForDSR = true;
+        (void)m_pVtEngine->_ListenForDSR();
+        (void)m_pVtEngine->RequestCursor();
+    }
+    return hr;
+}
+
+[[nodiscard]] HRESULT VtApiRoutines::ReadConsoleWImpl(IConsoleInputObject& context,
+                                                      gsl::span<char> buffer,
+                                                      size_t& written,
+                                                      std::unique_ptr<IWaitRoutine>& waiter,
+                                                      const std::string_view initialData,
+                                                      const std::wstring_view exeName,
+                                                      INPUT_READ_HANDLE_DATA& readHandleState,
+                                                      const HANDLE clientHandle,
+                                                      const DWORD controlWakeupMask,
+                                                      DWORD& controlKeyState) noexcept
+{
+    const auto hr = m_pUsualRoutines->ReadConsoleWImpl(context, buffer, written, waiter, initialData, exeName, readHandleState, clientHandle, controlWakeupMask, controlKeyState);
+    // If we're about to tell the caller to wait, let's synchronize the cursor we have with what
+    // the terminal is presenting in case there's a cooked read going on.
+    // TODO: we only need to do this in cooked read mode.
+    if (clientHandle)
+    {
+        m_listeningForDSR = true;
+        (void)m_pVtEngine->_ListenForDSR();
+        (void)m_pVtEngine->RequestCursor();

```

Comment:
The way `InputStateMachineEngine` and `VtInputThread` had originally handled this, it was a single time use flag set on creation of the `InputStateMachineEngine` to handle cursor inheritance at startup. But I've piggy-backed on that to ask the question again later. I suppose I could consolidate the two, but at the time, I really didn't want to tamper with the existing mechanics of ConPTY startup, so I left it separate for now in case rearranging it had a timing issue that was a headache. I could write a follow up in #10001 so I don't screw it up now with just adding the FHL code.